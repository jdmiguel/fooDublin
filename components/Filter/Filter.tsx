import React, { useReducer, Dispatch } from 'react';
import styled from 'styled-components';

type FilterData = {
  text: string;
  sort: 'cost' | 'rating';
  order: 'asc' | 'desc';
  id: number;
};

type FilterDataItemTypeWithIsActive = FilterData & { isActive: boolean };

type FilterProps = {
  className?: string;
  onSelect: (sort: string, order: string) => void;
  data: FilterData[];
};

type DataAction = { type: 'select'; id: number } | { type: 'clear' };

const StyledFilter = styled.div`
  width: 100%;
  max-width: 71rem;
  padding: 20px 0 10px;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  background-color: ${(props) => props.theme.palette.LIGHT_MAX};
  border-bottom: 1px solid ${(props) => props.theme.palette.LIGHT_SOFT};
  @media only screen and (min-width: 480px) {
    padding: 20px 20px 10px;
  }
  @media only screen and (min-width: 768px) {
    padding: 25px 0;
    justify-content: space-evenly;
  }
  @media only screen and (min-width: 992px) {
    border: 1px solid ${(props) => props.theme.palette.LIGHT_SOFT};
    border-radius: 4px;
  }
`;

const StyledFilterItem = styled.button<{ isActive: boolean }>`
  width: 144px;
  padding: 15px 8px;
  margin: 0 8px 15px;
  border: 1px solid ${(props) => props.theme.palette.DARK_MIN};
  background-color: ${(props) => props.theme.palette.LIGHT_MEDIUM};
  font-size: 0.9rem;
  color: ${(props) => props.theme.palette.DARK_MEDIUM};
  outline: none;
  cursor: pointer;
  background-color: ${({ theme, isActive }) =>
    `${isActive ? theme.palette.PRIMARY_LIGHT : theme.palette.LIGHT_MEDIUM}`};
  @media only screen and (min-width: 400px) {
    width: 180px;
  }
  @media only screen and (min-width: 480px) {
    width: 200px;
    margin: 0 10px 15px;
  }
  @media only screen and (min-width: 680px) {
    width: 250px;
  }
  @media only screen and (min-width: 768px) {
    width: auto;
    margin: 0;
    padding: 15px;
    font-size: 1rem;
  }
  @media only screen and (min-width: 992px) {
    padding: 20px;
  }
`;

const dataReducer = (data: FilterData[], action: DataAction) => {
  switch (action.type) {
    case 'select':
      return data.map((dataItem) => ({
        ...dataItem,
        isActive: action.id === dataItem.id,
      }));
    case 'clear':
      return data.map((dataItem) => ({
        ...dataItem,
        isActive: false,
      }));
  }
};

const Filter: React.FC<FilterProps> = ({ className, onSelect, data }) => {
  const dataWithIsActiveProp = data.map((dataItem) => ({
    ...dataItem,
    isActive: false,
  }));
  const [dataState, dispatch]: [
    FilterDataItemTypeWithIsActive[],
    Dispatch<DataAction>,
  ] = useReducer(dataReducer, dataWithIsActiveProp);

  const handleSelect = (id: number, sort: string, order: string) => {
    dispatch({ type: 'select', id });
    onSelect && onSelect(sort, order);
  };

  return (
    <StyledFilter className={className}>
      {dataState.map((item: FilterDataItemTypeWithIsActive) => (
        <StyledFilterItem
          isActive={item.isActive}
          key={item.id}
          onClick={(event: React.MouseEvent<HTMLElement>) => {
            event.preventDefault();
            handleSelect(item.id, item.sort, item.order);
          }}
        >
          {item.text}
        </StyledFilterItem>
      ))}
    </StyledFilter>
  );
};

export default Filter;
