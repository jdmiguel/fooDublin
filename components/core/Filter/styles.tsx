import styled from 'styled-components';

export const StyledFilter = styled.button<{ isActive: boolean }>`
  padding: 15px 8px;
  font-size: 0.9rem;
  border: 1px solid ${({ theme }) => theme.palette.LIGHT_SOFT};
  border-radius: 4px;
  color: ${({ theme }) => theme.palette.PRIMARY_MEDIUM};
  outline: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
  transition: all 0.2s ease-out;
  background-color: ${({ theme, isActive }) =>
    `${isActive ? theme.palette.PRIMARY_LIGHT : theme.palette.LIGHT_MEDIUM}`};
  @media only screen and (min-width: 768px) {
    width: auto;
    margin: 0;
    padding: 15px;
    font-size: 1rem;
  }
  @media only screen and (min-width: 1200px) {
    &:hover {
      background-color: ${({ theme }) => theme.palette.PRIMARY_LIGHT};
      color: ${({ theme }) => theme.palette.PRIMARY};
    }
  }
`;

export const StyledFilterText = styled.span`
  text-transform: uppercase;
  font-weight: 500;
`;

export const StyledFilterIcon = styled.i`
  font-size: 1.1rem;
  margin-right: 8px;
`;
