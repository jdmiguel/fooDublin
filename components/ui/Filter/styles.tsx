import styled from 'styled-components';

export const StyledFilterItem = styled.button<{ isActive: boolean }>`
  width: 144px;
  padding: 15px 8px;
  margin: 0 8px 15px;
  border: 1px solid ${({ theme }) => theme.palette.DARK_MIN};
  font-size: 0.9rem;
  color: ${({ theme }) => theme.palette.DARK_MEDIUM};
  outline: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  transition: background-color 0.2s ease-out;
  background-color: ${({ theme, isActive }) =>
    `${isActive ? theme.palette.PRIMARY_LIGHT : theme.palette.LIGHT_MIN}`};
  &:hover {
    background-color: ${({ theme }) => theme.palette.PRIMARY_LIGHT};
  }
  @media only screen and (min-width: 400px) {
    width: 180px;
  }
  @media only screen and (min-width: 480px) {
    width: 200px;
    margin: 0 10px 15px;
  }
  @media only screen and (min-width: 680px) {
    width: 250px;
  }
  @media only screen and (min-width: 768px) {
    width: auto;
    margin: 0;
    padding: 15px;
    font-size: 1rem;
  }
  @media only screen and (min-width: 992px) {
    padding: 20px;
  }
`;

export const StyledFilterPrimaryText = styled.span`
  text-transform: uppercase;
`;

export const StyledFilterSecondaryText = styled.span`
  display: none;
  @media only screen and (min-width: 920px) {
    display: block;
  }
`;

export const StyledFilterIcon = styled.i`
  font-size: 1.1rem;
  line-height: 18px;
  margin-left: 10px;
  display: block;
  @media only screen and (min-width: 920px) {
    display: none;
  }
`;
