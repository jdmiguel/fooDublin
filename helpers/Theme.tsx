import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
