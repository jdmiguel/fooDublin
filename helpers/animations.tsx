import { keyframes, css } from 'styled-components';

const fadeIn = (endValue: number) => keyframes`
  0% { opacity: 0 }
  100% { opacity: ${endValue} }
`;

const fadeInFromPosY = (initialPosY: number) => keyframes`
  0% { opacity: 0; transform: translateY(${initialPosY}px); }
  100% { opacity: 1; ; transform: translateY(0); }
`;

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const moveX = keyframes`
  0% { transform: translateX(-100%); }
  100% { transform: translateX(0); }
`;

export const fadeInAnimation = css<{ endValue?: number }>`
  animation: ${({ endValue }) => fadeIn(endValue || 1)} 0.15s linear;
`;

export const fadeInFromPosYAnimation = css<{
  initialPosY?: number;
  animationDuration?: number;
  animationDelay?: number;
}>`
  animation-name: ${({ initialPosY }) => fadeInFromPosY(initialPosY || 50)};
  animation-duration: ${({ animationDuration }) => animationDuration || 0.15}s;
  animation-delay: ${({ animationDelay }) => animationDelay || 0.25}s;
  animation-fill-mode: forwards;
  animation-timing-function: ease-out;
`;

export const circleLoaderAnimation = css`
  animation: ${rotate} 0.8s linear infinite;
`;

export const lineLoaderAnimation = css`
  animation: ${moveX} 1s ease-in-out infinite forwards;
`;
