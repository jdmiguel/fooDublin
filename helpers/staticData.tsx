import { CDN_URL_STATIC_DIRECTORY } from './utils';

export const DUBLIN_ID = 91;
export const BASE_URL = 'https://developers.zomato.com/api/v2.1/';

export const DEFAULT_TEXT_LOADING = 'Coming right up...';

export const MAX_RESTAURANT_DISPLAYED = 20;
export const MAX_RESTAURANT_RETRIEVED = 100;

export const MAX_SMALL_DEVICE_WIDTH = 640;

export const SCROLL_FACTOR = 4;
export const SCROLL_INITIAL_MOBILE_FACTOR = 10;
export const SCROLL_OFFSET_MOBILE_FACTOR = 4;
export const SCROLL_OFFSET_DESKTOP_FACTOR = 1.2;
export const SCROLL_DELAY = 4;

export const SHOWING_SCROLLUP_BUTTON_HEIGHT = 900;

export enum PlaceholderText {
  BLURRED = 'Search for locals...',
  FOCUSED = 'Start typing to search...',
}

export const DEFAULT_SUGGESTIONS = [
  {
    id: 0,
    imgSrc: '',
    firstText: '',
    secondText: '',
  },
];

export const LOCATIONS = [
  {
    id: 91,
    name: 'Dublin',
    path: 'dublin',
  },
  {
    id: 162239,
    name: 'South City West',
    path: 'south-city-west',
  },
  {
    id: 96201,
    name: 'Temple Bar',
    path: 'temple-bar',
  },
  {
    id: 95051,
    name: 'Clondalkin',
    path: 'clondalkin',
  },
  {
    id: 96181,
    name: 'Swords',
    path: 'swords',
  },
  {
    id: 96191,
    name: 'Tallaght',
    path: 'tallaght',
  },
  {
    id: 94881,
    name: 'Blanchardstown',
    path: 'blanchardstown',
  },
  {
    id: 162240,
    name: 'South City East',
    path: 'south-city-east',
  },
  {
    id: 95311,
    name: 'Dundrum',
    path: 'dundrum',
  },
  {
    id: 95341,
    name: 'Finglas',
    path: 'finglas',
  },
  {
    id: 94791,
    name: 'Ballyfermot',
    path: 'ballyfermot',
  },
  {
    id: 96081,
    name: 'Santry',
    path: 'santry',
  },
  {
    id: 95711,
    name: 'Lucan',
    path: 'lucan',
  },
  {
    id: 94671,
    name: 'Artane',
    path: 'artane',
  },
  {
    id: 95721,
    name: 'Malahide',
    path: 'malahide',
  },
  {
    id: 95301,
    name: 'Dun Laoghaire',
    path: 'dun-laoghaire',
  },
  {
    id: 96271,
    name: 'Walkinstown',
    path: 'walkinstown',
  },
  {
    id: 94871,
    name: 'Blackrock',
    path: 'blackrock',
  },
  {
    id: 95851,
    name: 'Palmerstown',
    path: 'palmerstown',
  },
  {
    id: 95981,
    name: 'Rathmines',
    path: 'rathmines',
  },
  {
    id: 94701,
    name: 'Balbriggan',
    path: 'balbriggan',
  },
  {
    id: 94761,
    name: 'Ballybrack',
    path: 'ballybrack',
  },
  {
    id: 96111,
    name: 'Smithfield',
    path: 'smithfield',
  },
  {
    id: 95181,
    name: 'Crumlin',
    path: 'crumlin',
  },
  {
    id: 94971,
    name: 'Castleknock',
    path: 'castleknock',
  },
  {
    id: 95511,
    name: 'Howth',
    path: 'howth',
  },
  {
    id: 94951,
    name: 'Cabra',
    path: 'cabra',
  },
  {
    id: 96291,
    name: 'Whitehall',
    path: 'whitehall',
  },
  {
    id: 95901,
    name: 'Portmarnock',
    path: 'portmarnock',
  },
  {
    id: 95641,
    name: 'Kimmage',
    path: 'kimmage',
  },
];

export const CUISINES = [
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/stew.svg`,
    id: 135,
    name: 'Irish',
    path: 'irish',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/pizza.svg`,
    id: 55,
    name: 'Italian',
    path: 'italian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/bruschetta.svg`,
    id: 70,
    name: 'Mediterranean',
    path: 'mediterranean',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rice.svg`,
    id: 3,
    name: 'Asian',
    path: 'asian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/healthy_food.svg`,
    id: 143,
    name: 'Healthy Food',
    path: 'healthy-food',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/gulab.svg`,
    id: 148,
    name: 'Indian',
    path: 'indian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/taco.svg`,
    id: 73,
    name: 'Mexican',
    path: 'mexican',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/fries.svg`,
    id: 40,
    name: 'Fast Food',
    path: 'fast-food',
  },
];

export const HIGHLIGHTED_RESTAURANTS = [
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/cleaver-east.jpg`,
    id: 9100233,
    path: 'cleaver-east',
    name: 'Cleaver East Restaurant',
    description:
      'Situated in the heart of Ireland’s capital city, on East Essex Street in the popular Temple Bar area, Cleaver East Restaurant offers the finest Irish and European cuisine.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/tapas-de-lola.jpg`,
    id: 16518539,
    path: 'tapas-de-lola',
    name: 'Tapas de Lola',
    description:
      'This restaurant is a love letter to all things Spanish and flavorful. Its owners make regular pilgrimages to Spain for inspiration and pack it all into a lively menu of small plates.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/ely-wine.jpg`,
    id: 16517305,
    path: 'ely-wine',
    name: 'Ely Wine Bar',
    description:
      'Eric and his wife Michelle are the founders of this clubby place where you can eat and drink well both downstairs in the cozy basement and upstairs in an elegant dining room.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/oxmantown.jpg`,
    id: 16520229,
    path: 'oxmantown',
    name: 'Oxmantown',
    description:
      'In this small place, the art of a brilliant sandwich means every element is taken seriously, like the breakfast sandwich that comes with butcher Jack McCarthy’s black pudding.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/chapter-one.jpg`,
    id: 9100702,
    path: 'chapter-one',
    name: 'Chapter One',
    description:
      'Dublin writers are more closely associated with booze rather than food, but Chapter One is a Michelin-starred restaurant in the basement of the Dublin Writers Museum.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rosa-madre.jpg`,
    id: 9101166,
    path: 'rosa-madre',
    name: 'Rosa Madre',
    description:
      'In the menu-hawking clamor of Temple Bar you might long for a quieter restaurant where you won’t feel like a tourist on a conveyor belt. If so, Rosa Madre is your place.',
  },
];

export const HIGHLIGHT_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/highlight-generic.png`;
export const THUMB_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-thumb.png`;
export const DETAIL_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-detail.png`;
export const DEFAULT_RATING_STAR_LIST = [
  { id: 1, type: 'star_outline' },
  { id: 2, type: 'star_outline' },
  { id: 3, type: 'star_outline' },
  { id: 4, type: 'star_outline' },
  { id: 5, type: 'star_outline' },
];

export enum FilterSort {
  COST = 'cost',
  RANK = 'rating',
}
export enum FilterOrder {
  ASC = 'asc',
  DESC = 'desc',
}

export const FILTER_DATA = [
  {
    primaryText: 'cost',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.COST,
    order: FilterOrder.ASC,
    id: 1,
  },
  {
    primaryText: 'cost',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.COST,
    order: FilterOrder.DESC,
    id: 2,
  },
  {
    primaryText: 'rank',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.RANK,
    order: FilterOrder.ASC,
    id: 3,
  },
  {
    primaryText: 'rank',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.RANK,
    order: FilterOrder.DESC,
    id: 4,
  },
];
