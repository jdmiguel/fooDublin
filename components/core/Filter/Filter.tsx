import { StyledFilter, StyledFilterText, StyledFilterIcon } from './styles';
import { FilterData, FilterType } from '../types';

type FilterProps = {
  className?: string;
  data: FilterData;
  onClick: (filter: FilterType) => void;
};

export const Filter: React.FC<FilterProps> = ({
  data: { isActive, text, icon, type },
  onClick,
}) => (
  <StyledFilter isActive={isActive} onClick={() => onClick(type)}>
    <StyledFilterText>{text}</StyledFilterText>
    <StyledFilterIcon className="material-icons">{icon}</StyledFilterIcon>
  </StyledFilter>
);
