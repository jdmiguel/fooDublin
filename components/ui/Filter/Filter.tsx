import React, { useReducer, Dispatch } from 'react';
import styled from 'styled-components';

import { FilterSort, FilterOrder } from '../../../helpers/staticData';

type FilterData = {
  primaryText: string;
  secondaryText: string;
  icon: string;
  sort: FilterSort.COST | FilterSort.RANK;
  order: FilterOrder.ASC | FilterOrder.DESC;
  id: number;
};

type FilterDataItemTypeWithIsActive = FilterData & { isActive: boolean };

type FilterProps = {
  className?: string;
  onClick: (sort: string, order: string) => void;
  data: FilterData[];
};

type DataAction = { type: 'select'; id: number } | { type: 'clear' };

const StyledFilterItem = styled.button<{ isActive: boolean }>`
  width: 144px;
  padding: 15px 8px;
  margin: 0 8px 15px;
  border: 1px solid ${({ theme }) => theme.palette.DARK_MIN};
  font-size: 0.9rem;
  color: ${({ theme }) => theme.palette.DARK_MEDIUM};
  outline: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  transition: background-color 0.2s ease-out;
  background-color: ${({ theme, isActive }) =>
    `${isActive ? theme.palette.PRIMARY_LIGHT : theme.palette.LIGHT_MIN}`};
  &:hover {
    background-color: ${({ theme }) => theme.palette.PRIMARY_LIGHT};
  }
  @media only screen and (min-width: 400px) {
    width: 180px;
  }
  @media only screen and (min-width: 480px) {
    width: 200px;
    margin: 0 10px 15px;
  }
  @media only screen and (min-width: 680px) {
    width: 250px;
  }
  @media only screen and (min-width: 768px) {
    width: auto;
    margin: 0;
    padding: 15px;
    font-size: 1rem;
  }
  @media only screen and (min-width: 992px) {
    padding: 20px;
  }
`;

const StyledFilterPrimaryText = styled.span`
  text-transform: uppercase;
`;

const StyledFilterSecondaryText = styled.span`
  display: none;
  @media only screen and (min-width: 920px) {
    display: block;
  }
`;

const StyledFilterIcon = styled.i`
  font-size: 1.1rem;
  line-height: 18px;
  margin-left: 10px;
  display: block;
  @media only screen and (min-width: 920px) {
    display: none;
  }
`;

const dataReducer = (data: FilterData[], action: DataAction) => {
  switch (action.type) {
    case 'select':
      return data.map((dataItem) => ({
        ...dataItem,
        isActive: action.id === dataItem.id,
      }));
    case 'clear':
      return data.map((dataItem) => ({
        ...dataItem,
        isActive: false,
      }));
  }
};

const Filter: React.FC<FilterProps> = ({ onClick, data }) => {
  const dataWithIsActiveProp = data.map((dataItem) => ({
    ...dataItem,
    isActive: false,
  }));
  const [dataState, dispatch]: [
    FilterDataItemTypeWithIsActive[],
    Dispatch<DataAction>,
  ] = useReducer(dataReducer, dataWithIsActiveProp);

  const handleClick = (
    isActive: boolean,
    id: number,
    sort: string,
    order: string,
  ) => {
    dispatch(!isActive ? { type: 'select', id } : { type: 'clear' });
    onClick(!isActive ? sort : '', !isActive ? order : '');
  };

  return (
    <div className="grid-x grid-margin-x">
      {dataState.map((item) => (
        <StyledFilterItem
          isActive={item.isActive}
          className="cell small-6 medium-3 large-3"
          key={item.id}
          onClick={() => {
            handleClick(item.isActive, item.id, item.sort, item.order);
          }}
        >
          <StyledFilterPrimaryText>{item.primaryText}</StyledFilterPrimaryText>
          <StyledFilterSecondaryText>
            {` ${item.secondaryText}`}
          </StyledFilterSecondaryText>
          <StyledFilterIcon className="material-icons">
            {item.icon}
          </StyledFilterIcon>
        </StyledFilterItem>
      ))}
    </div>
  );
};

export default Filter;
