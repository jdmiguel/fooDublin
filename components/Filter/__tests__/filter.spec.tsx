import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import Filter from '../Filter';

import { renderWithTheme } from '../../../helpers/Theme';
import { FILTER_DATA } from '../../../helpers/staticData';

describe('Component: Filter', () => {
  it('should render', () => {
    const { container } = render(
      renderWithTheme(<Filter onSelect={() => {}} data={FILTER_DATA} />),
    );

    expect(container.firstChild).toMatchSnapshot();
  });
  it('should active filter and call function on click any filter', () => {
    const handleSelect = jest.fn();
    const { container } = render(
      renderWithTheme(<Filter onSelect={handleSelect} data={FILTER_DATA} />),
    );
    const filter = container.firstChild as HTMLDivElement;
    const firstFilterItem = filter.querySelector('button');
    const secondFilterItem = filter.querySelector('button:nth-of-type(2)');
    const thirdFilterItem = filter.querySelector('button:nth-of-type(3)');
    const fourthFilterItem = filter.querySelector('button:nth-of-type(4)');

    // check background color of first filterItem and rest filterItems
    expect(firstFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(secondFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(thirdFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(fourthFilterItem).toHaveStyleRule('background-color', '#F3F3F3');

    fireEvent.click(firstFilterItem);

    //  check background color of first filterItem and rest filterItems and call callback
    expect(firstFilterItem).toHaveStyleRule('background-color', '#FDF3E5');
    expect(secondFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(thirdFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(fourthFilterItem).toHaveStyleRule('background-color', '#F3F3F3');
    expect(handleSelect).toHaveBeenCalledTimes(1);
  });
});
