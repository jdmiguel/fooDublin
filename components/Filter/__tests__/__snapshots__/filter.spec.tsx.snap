// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Filter should render 1`] = `
.c0 {
  background-color: #FFFFFF;
  border-bottom: 1px solid #D8D8D8;
}

.c1 {
  width: 144px;
  padding: 15px 8px;
  margin: 0 8px 15px;
  border: 1px solid #89959B;
  background-color: #F3F3F3;
  font-size: 0.9rem;
  color: #585858;
  outline: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #F3F3F3;
}

.c2 {
  text-transform: uppercase;
}

.c3 {
  display: none;
}

.c4 {
  font-size: 1.1rem;
  line-height: 18px;
  margin-left: 10px;
  display: block;
}

@media only screen and (min-width:480px) {
  .c0 {
    padding: 20px 20px 10px;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding: 20px;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    border: 1px solid #D8D8D8;
    border-radius: 4px;
  }
}

@media only screen and (min-width:400px) {
  .c1 {
    width: 180px;
  }
}

@media only screen and (min-width:480px) {
  .c1 {
    width: 200px;
    margin: 0 10px 15px;
  }
}

@media only screen and (min-width:680px) {
  .c1 {
    width: 250px;
  }
}

@media only screen and (min-width:768px) {
  .c1 {
    width: auto;
    margin: 0;
    padding: 15px;
    font-size: 1rem;
  }
}

@media only screen and (min-width:992px) {
  .c1 {
    padding: 20px;
  }
}

@media only screen and (min-width:920px) {
  .c3 {
    display: block;
  }
}

@media only screen and (min-width:920px) {
  .c4 {
    display: none;
  }
}

<div
  class="c0  grid-container"
>
  <div
    class="grid-x grid-margin-x"
  >
    <button
      class="c1 cell small-6 medium-3 large-3"
    >
      <span
        class="c2"
      >
        cost
      </span>
      <span
        class="c3"
      >
         - high to low
      </span>
      <i
        class="c4 material-icons"
      >
        keyboard_arrow_up
      </i>
    </button>
    <button
      class="c1 cell small-6 medium-3 large-3"
    >
      <span
        class="c2"
      >
        cost
      </span>
      <span
        class="c3"
      >
         - low to high
      </span>
      <i
        class="c4 material-icons"
      >
        keyboard_arrow_down
      </i>
    </button>
    <button
      class="c1 cell small-6 medium-3 large-3"
    >
      <span
        class="c2"
      >
        rank
      </span>
      <span
        class="c3"
      >
         - high to low
      </span>
      <i
        class="c4 material-icons"
      >
        keyboard_arrow_up
      </i>
    </button>
    <button
      class="c1 cell small-6 medium-3 large-3"
    >
      <span
        class="c2"
      >
        rank
      </span>
      <span
        class="c3"
      >
         - low to high
      </span>
      <i
        class="c4 material-icons"
      >
        keyboard_arrow_down
      </i>
    </button>
  </div>
</div>
`;
